backend/entities/Address.cs

```diff
@@ -68,14 +63,14 @@ public class Address : BaseEntity
         /// </summary>
         /// <param name="address1"></param>
         /// <param name="address2"></param>
-        /// <param name="cityId"></param>
+        /// <param name="administrativeArea"></param>
         /// <param name="provinceId"></param>
         /// <param name="postal"></param>
-        public Address(string address1, string address2, int cityId, string provinceId, string postal)
+        public Address(string address1, string address2, string administrativeArea, string provinceId, string postal)
         {
             this.Address1 = address1;
             this.Address2 = address2;
-            this.CityId = cityId;
+            this.AdministrativeArea = administrativeArea ?? throw new ArgumentNullException(nameof(administrativeArea));
```

Comment:
it seems weird to do the enforcement at this level no?

Reply:
A standard practice in constructors to provide immediate feedback on required arguments.  I should actually enforce more of them as it is weird it is the only one.